packages/base/matrix-room.gts

```diff
@@ -0,0 +1,415 @@
+import {
+  contains,
+  containsMany,
+  field,
+  Component,
+  Card,
+  primitive,
+  CardBase,
+  createFromSerialized,
+} from './card-api';
+import StringCard from './string';
+import DateTimeCard from './datetime';
+import IntegerCard from './integer';
+import MarkdownCard from './markdown';
+import { BoxelMessage } from '@cardstack/boxel-ui';
+import cssVar from '@cardstack/boxel-ui/helpers/css-var';
+import { formatRFC3339 } from 'date-fns';
+import Modifier from 'ember-modifier';
+import { type LooseSingleCardDocument } from '@cardstack/runtime-common';
+
+const roomMembers = new Map<string, RoomMemberCard>();
+
+// this is so we can have triple equals equivalent room member cards
+function upsertRoomMember(
+  userId: string,
+  displayName?: string
+): RoomMemberCard {
+  let member = roomMembers.get(userId);
+  if (!member) {
+    member = new RoomMemberCard({ userId });
+    roomMembers.set(userId, member);
+  }
+
+  // patch in the display name in case we don't have one yet
+  if (displayName && member.displayName === undefined) {
+    member.displayName = displayName;
+  }
```

Comment:
Is there a case where a member's displayName can change?

Reply:
yes--there should be a matrix event that can be sent in a room when a user's display name changes. I haven't tracked that event down though...